### DISCUSSION TITLE
About LMP

### DISCUSSION THREAD

**Discussion by qwerty12345697** [Anura3D/Anura3D_OpenSource#178] [Q&A]
In this article (Simulation of Soil Fluidization Around a Pressurized Leaking Pipe Using the Material Point Method), in the simulation process of the two-phase two-point material point method, some locations will continue to generate water substance points, and some will continue to eliminate water substance points. If so, how do you implement control over the precise location of the point of matter in your code?
Moreover, if the biphasic and double-point method is used in the process of simulated rainfall, how should the material point of the rainfall water be realized? How can we simulate the point of matter in the effluent without using infiltration?

**Comment by murphy-john** [comment]
The inflow and outflow boundary conditions haven't been implemented in the open source version of the code. The article you are referring to used a branch of the code where the inflows/outflows were coded.

**Comment by nankemengyu1** [comment]
> In this article (Simulation of Soil Fluidization Around a Pressurized Leaking Pipe Using the Material Point Method), in the simulation process of the two-phase two-point material point method, some locations will continue to generate water substance points, and some will continue to eliminate water substance points. If so, how do you implement control over the precise location of the point of matter in your code? Moreover, if the biphasic and double-point method is used in the process of simulated rainfall, how should the material point of the rainfall water be realized? How can we simulate the point of matter in the effluent without using infiltration?

Hello, qwerty
  Have you implemented the function of continuous generation and elimination of water material points like this article? I am also facing such a situation, but I do not know how to modify the source code to have such a function, if you have implemented it, can you share your experience with me? Thank you very much!
